src/OrleansTestKit/TestGrainFactory.cs

```diff
@@ -100,8 +99,14 @@ internal Mock<T> AddProbe<T>(IGrainIdentity identity, string classPrefix = null)
             return mock;
         }
 
+        internal void AddProbe<T>(Func<IGrainIdentity, T> factory) where T : class, IGrain => _probeFactories.Add(typeof(T), factory);
+
         internal void AddProbe<T>(Func<IGrainIdentity, IMock<T>> factory)
-            where T : class, IGrain => _probeFactories.Add(typeof(T), factory);
+            where T : class, IGrain
+        {
+            var adaptedFactory = new Func<IGrainIdentity, T>(grainIdentity => factory(grainIdentity).Object);
```

Comment:
The (removed) line 139 below would've tolerated a user-provided function that returned `null`.

```cs
grain = mock?.Object;
```

A user shouldn't do this, but to preserve existing behaviors we should pass through the `null` value.

```suggestion
            var adaptedFactory = new Func<IGrainIdentity, T>(grainIdentity => factory(grainIdentity)?.Object);
```

Reply:
of course, this is doe